### PR TITLE
feat(module:modal): support draggable modal

### LINKED ISSUE #4549: ModalComponents不支持header drag
## What problem does this feature solve?
全局模态框支持drag。目前是通过引入angular cdk drag，通过[title]传入方式来支持drag。但是这样重复代码太多，每一个模态框都需要复写title模板，是否能够提供一些方式解决以上需求。
1,默认支持drag，全局配置即可实现
2，支持配置全局默认的modal title，自行引入angular cdk drag来支持drag

## What does the proposed API look like?
[dragable]=""

### PR DESCRIPTION
**Pull request by hsuanxyz**
test(module:modal): add draggable test case

close #4549, cc #2705, #1904, #120

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #4549, #2705, #1904, #120

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by wendellhu95** [comment]
Confirm 这种简单样式的 modal 应该需要改变鼠标样式。

遮罩顺序有点问题。

图片

**Comment by hsuanxyz** [comment]
@wendellhu95 DEMO 里面有对于代码，现在 modal 不处理这些，只管挂载拖拽节点

File: components/modal/demo/draggable.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
+/* declarations: NzModalDraggableComponent */
+
+import { Component } from '@angular/core';
+import { NzModalService } from 'ng-zorro-antd';
+
+@Component({
+  selector: 'nz-demo-modal-draggable',
+  template: `
+    <button nz-button [nzType]="'primary'" (click)="showModal()">Show Modal</button>
+    &nbsp;
```
Comment (vthinkxie) [inline components/modal/demo/draggable.ts:10]:
```suggestion
```
should not including &nbsp in demo

File: components/modal/demo/draggable.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
+/* declarations: NzModalDraggableComponent */
```
Comment (vthinkxie) [inline components/modal/demo/draggable.ts:1]:
```suggestion
```

**Comment by sawyerbutton** [comment]
Hope this PR will be merged ASAP.

**Comment by 100cm** [comment]
希望尽快merge

**Comment by jongomes** [comment]
need feature

**Comment by sevenflip** [comment]
@hsuanxyz @hullis is there a reason why this PR is not yet merged?

**Comment by HyperLife1119** [comment]
Close via #8419.

**Pull request closed without merging**